presto-spi/src/main/java/com/facebook/presto/spi/type/UnscaledDecimal128Arithmetic.java

```diff
@@ -730,6 +730,117 @@ static void shiftRight(Slice decimal, int rightShifts, boolean roundUp, Slice re
         pack(result, lo, hi, negative);
     }
 
+    public static Slice divideRoundUp(long dividend, int dividendScaleFactor, long divisor)
+    {
+        return divideRoundUp(
+                Math.abs(dividend), dividend < 0 ? SIGN_LONG_MASK : 0, dividendScaleFactor,
+                Math.abs(divisor), divisor < 0 ? SIGN_LONG_MASK : 0);
+    }
+
+    public static Slice divideRoundUp(long dividend, int dividendScaleFactor, Slice divisor)
+    {
+        return divideRoundUp(
+                Math.abs(dividend), dividend < 0 ? SIGN_LONG_MASK : 0, dividendScaleFactor,
+                getRawLong(divisor, 0), getRawLong(divisor, 1));
+    }
+
+    public static Slice divideRoundUp(Slice dividend, int dividendScaleFactor, long divisor)
+    {
+        return divideRoundUp(
+                getRawLong(dividend, 0), getRawLong(dividend, 1), dividendScaleFactor,
+                Math.abs(divisor), divisor < 0 ? SIGN_LONG_MASK : 0);
+    }
+
+    public static Slice divideRoundUp(Slice dividend, int dividendScaleFactor, Slice divisor)
+    {
+        return divideRoundUp(
+                getRawLong(dividend, 0), getRawLong(dividend, 1), dividendScaleFactor,
+                getRawLong(divisor, 0), getRawLong(divisor, 1));
+    }
+
+    private static Slice divideRoundUp(long dividendLow, long dividendHi, int dividendScaleFactor, long divisorLow, long divisorHi)
```

Comment:
`Hi -> High`

Karol has some pull request that fixes that in rest of the code, so don't introduce new code with abbreviations